src/flag_gems/ops/uniform.py

```diff
@@ -0,0 +1,77 @@
+import logging
+
+import torch
+import triton
+import triton.language as tl
+
+from flag_gems.utils.random_utils import philox_cuda_seed_offset, uint_to_uniform_float
+from flag_gems.utils.shape_utils import volume
+
+
+def heur_block(args):
+    if args["N"] <= 512:
+        return 512
+    else:
+        return 1024
+
+
+def heur_num_warps(args):
+    if args["N"] <= 512:
+        return 4
+    elif args["N"] <= 1024:
+        return 8
+    else:
+        return 16
+
+
+@triton.heuristics(
+    {
+        "BLOCK": heur_block,
+        "num_warps": heur_num_warps,
+    }
+)
+@triton.jit(do_not_specialize=["philox_seed", "philox_offset"])
+def uniform_kernel(
+    out_ptr,
+    N,
+    philox_seed,
+    philox_offset,
+    from_,
+    to,
+    BLOCK: tl.constexpr,
+):
+    philox_seed = philox_seed.to(tl.int64)
+    philox_offset = philox_offset.to(tl.int64)
+    c0 = (philox_offset & 0xFFFFFFFF).to(tl.uint32)
+    c1 = ((philox_offset >> 32) & 0xFFFFFFFF).to(tl.uint32)
+    i4 = tl.program_id(0) * BLOCK + tl.arange(0, BLOCK)
+    c0 += i4
+    _O = c0 * 0
+    r0, r1, r2, r3 = tl.philox(philox_seed, c0, c1, _O, _O)
+    r0 = uint_to_uniform_float(r0) * (to - from_) + from_
+    r1 = uint_to_uniform_float(r1) * (to - from_) + from_
+    r2 = uint_to_uniform_float(r2) * (to - from_) + from_
+    r3 = uint_to_uniform_float(r3) * (to - from_) + from_
```

Comment:
We need to outline this chunk of code into a function in random_utils.

Reply:
triton does not support high-order function, use tow function will cause performance decrease